File: Sem3/Advanced Methods of Programming/MyInterpreter - Java Console Application/project/src/view/RunExample.java
package view;


import controller.Controller;
import controller.ControllerException;


public class RunExample extends Command
{
    private Controller controller;


    public RunExample(String key, String description, Controller controller)
    {
        super(key, description);
        this.controller = controller;
    }


    @Override
    public void execute()
    {
        try
        {
            controller.allSteps();
            System.out.println("Command executed successfully.");
        }
        catch (ControllerException | InterruptedException exception)
        {
            System.out.println(exception.getMessage());
        }
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/view/RunExample.cs
using System;
using MyInterpreter_CSharp.controller;

namespace MyInterpreter_CSharp.view
{
    public class RunExample : Command
    {
        private Controller _controller;
        public RunExample(string key, string description, Controller controller) : base(key, description)
        {
            _controller = controller;
        }

        public override void Execute()
        {
            try
            {
                _controller.AllStep();
                Console.WriteLine("Command executed successfully.");
            }
            catch (ControllerException exception)
            {
                Console.WriteLine(exception.Message);
            }
        }
    }
}